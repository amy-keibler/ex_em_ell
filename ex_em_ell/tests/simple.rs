#[derive(Debug, PartialEq, ex_em_ell::ToXmlDocument)]
struct Example {
    field: String,
}

#[test]
fn test_from_xml() {
    let input = "<example><field>value</field></example>";
    let expected = Example {
        field: "value".to_string(),
    };
}

#[test]
fn test_to_xml() {
    let input = Example {
        field: "value".to_string(),
    };
    let actual = ex_em_ell::to_string_pretty(&input).expect("Failed to output XML");
    insta::assert_snapshot!(actual);
}
