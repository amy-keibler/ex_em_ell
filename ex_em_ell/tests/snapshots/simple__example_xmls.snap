---
source: ex_em_ell/tests/simple.rs
expression: round_trip
input_file: ex_em_ell/tests/data/simple/valid_example.xml
---
<?xml version="1.0" encoding="utf-8"?>
<example>
  <field>value</field>
  <child>
    <field>value</field>
  </child>
</example>
